src/uu/dircolors/src/dircolors.rs

```diff
@@ -52,28 +64,78 @@ pub fn guess_syntax() -> OutputFmt {
         .accept_any();
 
-    let matches = app!(SYNTAX, SUMMARY, LONG_HELP)
-        .optflag("b", "sh", "output Bourne shell code to set LS_COLORS")
-        .optflag(
-            "",
-            "bourne-shell",
-            "output Bourne shell code to set LS_COLORS",
+    let usage = get_usage();
+
+    /* Clap has .visible_alias, but it generates help like this
+     *     -b, --sh                output Bourne shell code to set LS_COLORS [aliases: bourne-shell]
+     * whereas we want help like this
+     *     -b, --sh                output Bourne shell code to set LS_COLORS
+     *         --bourne-shell      output Bourne shell code to set LS_COLORS
+     * (or preferably like the original, but that doesn't seem possible with clap:)
+     *     -b, --sh, --bourne-shell    output Bourne shell code to set LS_COLORS
+     * therefore, command aliases are defined manually as multiple commands
+     */
+    let matches = App::new(executable!())
+        .version(crate_version!())
+        .about(SUMMARY)
+        .usage(&usage[..])
+        .after_help(LONG_HELP)
+        .arg(
+            Arg::with_name(options::SH)
+                .long("sh")
+                .short("b")
+                .help("output Bourne shell code to set LS_COLORS")
+                .display_order(1),
         )
-        .optflag("c", "csh", "output C shell code to set LS_COLORS")
-        .optflag("", "c-shell", "output C shell code to set LS_COLORS")
-        .optflag("p", "print-database", "print the byte counts")
-        .parse(args);
-
-    if (matches.opt_present("csh")
-        || matches.opt_present("c-shell")
-        || matches.opt_present("sh")
-        || matches.opt_present("bourne-shell"))
-        && matches.opt_present("print-database")
+        .arg(
+            Arg::with_name(options::BOURNE_SHELL)
+                .long("bourne-shell")
+                .help("output Bourne shell code to set LS_COLORS")
+                .display_order(2),
+        )
+        .arg(
+            Arg::with_name(options::CSH)
+                .long("csh")
+                .short("c")
+                .help("output C shell code to set LS_COLORS")
+                .display_order(3),
+        )
+        .arg(
+            Arg::with_name(options::C_SHELL)
+                .long("c-shell")
+                .help("output C shell code to set LS_COLORS")
+                .display_order(4),
+        )
+        .arg(
+            Arg::with_name(options::PRINT_DATABASE)
+                .long("print-database")
+                .short("p")
+                .help("print the byte counts")
+                .display_order(5),
+        )
+        .arg(Arg::with_name(options::FILE).hidden(true).multiple(true))
+        .get_matches_from(&args);
+
+    let files = matches
+        .values_of(options::FILE)
+        .map_or(vec![], |file_values| file_values.collect());
+
+    // clap provides .conflicts_with / .conflicts_with_all, but we want to
```

Comment:
This is fine for now. I think we should start making functions in `uucore` that convert `clap` errors to a more GNU-like format, but those don't exist yet so this is fine.